### PR TITLE
Add minor lint: inherent_method_no_longer_unsafe

### PR DESCRIPTION
**Pull request by MarcosAsh**
Adds a new Minor/Allow lint that detects when a public inherent method or associated fn is no longer unsafe. From the tracking list in #949.

The query mirrors inherent_method_unsafe_added with the unsafe filters inverted, using the same 3-fold impl pattern to handle trait migration and method removal edge cases.

Test crate covers associated fn, self method, and enum associated fn becoming safe, plus negative cases for stays-unsafe, already-safe, removed-entirely, and private-struct.

### CODE REVIEW

File: src/lints/inherent_method_no_longer_unsafe.ron [hunk 1]
```diff
@@ -0,0 +1,105 @@
+SemverQuery(
+    id: "inherent_method_no_longer_unsafe",
+    human_readable_name: "pub method is no longer unsafe",
+    description: "A method or associated fn is no longer unsafe to call. Callers that use an `unsafe` block may get `unused_unsafe` warnings.",
+    required_update: Minor,
+    lint_level: Allow,
+    reference_link: Some("https://doc.rust-lang.org/book/ch19-01-unsafe-rust.html#calling-an-unsafe-function-or-method"),
```
Comment (obi1kenobi) [inline src/lints/inherent_method_no_longer_unsafe.ron:7]: This doc appears to have moved to a new address recently. Looks like a chapter was added to the book.
```suggestion
    reference_link: Some("https://doc.rust-lang.org/book/ch20-01-unsafe-rust.html#calling-an-unsafe-function-or-method"),
```

**Comment by obi1kenobi** [comment]
Thanks for the PR!
